### ISSUE TITLE
Can not start workspace - Forbidden! Configured service account doesn't have access

### ISSUE THREAD

**Issue by ibuziuk** [openshiftio/openshift.io#1281]
che: 2.0.45
Error while starting a workspace:

```
2017-11-02 21:44:02,503[aceSharedPool-0]  [INFO ] [o.e.c.p.o.c.OpenShiftConnector 538]  - Error while creating Pod, removing deployment
--
  | 2017-11-02 21:44:02,504[aceSharedPool-0]  [INFO ] [o.e.c.p.o.c.OpenShiftConnector 539]  - Failure executing: POST at: https://kubernetes.default.svc/api/v1/namespaces/ibuziuk-che/services. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked..
  | 2017-11-02 21:44:02,575[aceSharedPool-0]  [WARN ] [p.o.c.k.KubernetesResourceUtil 42]   - No Deployment with name che-ws-3v6d5ms6stnn6kds could be found
  | 2017-11-02 21:44:02,653[aceSharedPool-0]  [WARN ] [p.o.c.k.KubernetesResourceUtil 62]   - No Service with selector deployment=che-ws-3v6d5ms6stnn6kds could be found
  | 2017-11-02 21:44:02,736[aceSharedPool-0]  [WARN ] [p.o.c.k.KubernetesResourceUtil 77]   - No Route with label deployment=che-ws-3v6d5ms6stnn6kds could be found
  | 2017-11-02 21:44:02,748[aceSharedPool-0]  [ERROR] [o.e.c.a.w.s.WorkspaceManager 683]    - No Route with label deployment=che-ws-3v6d5ms6stnn6kds could be found
  | org.eclipse.che.api.core.ServerException: No Route with label deployment=che-ws-3v6d5ms6stnn6kds could be found
  | at org.eclipse.che.api.environment.server.CheEnvironmentEngine.startInstance(CheEnvironmentEngine.java:1002)
  | at org.eclipse.che.api.environment.server.CheEnvironmentEngine.startEnvironmentQueue(CheEnvironmentEngine.java:794)
  | at org.eclipse.che.api.environment.server.CheEnvironmentEngine.start(CheEnvironmentEngine.java:262)
  | at org.eclipse.che.api.workspace.server.WorkspaceRuntimes.startEnvironmentAndPublishEvents(WorkspaceRuntimes.java:694)
  | at org.eclipse.che.api.workspace.server.WorkspaceRuntimes.access$100(WorkspaceRuntimes.java:103)
  | at org.eclipse.che.api.workspace.server.WorkspaceRuntimes$StartTask.call(WorkspaceRuntimes.java:986)
  | at org.eclipse.che.api.workspace.server.WorkspaceRuntimes$StartTask.call(WorkspaceRuntimes.java:948)
  | at org.eclipse.che.commons.lang.concurrent.CopyThreadLocalCallable.call(CopyThreadLocalCallable.java:30)
  | at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  | at java.lang.Thread.run(Thread.java:748)
  | Caused by: org.eclipse.che.api.core.ServerException: No Route with label deployment=che-ws-3v6d5ms6stnn6kds could be found
  | at org.eclipse.che.plugin.docker.machine.MachineProviderImpl.startService(MachineProviderImpl.java:396)
  | at org.eclipse.che.api.environment.server.CheEnvironmentEngine.lambda$startEnvironmentQueue$6(CheEnvironmentEngine.java:765)
  | at org.eclipse.che.api.environment.server.CheEnvironmentEngine.startInstance(CheEnvironmentEngine.java:939)
  | ... 11 common frames omitted
  | Caused by: java.io.IOException: No Route with label deployment=che-ws-3v6d5ms6stnn6kds could be found
  | at org.eclipse.che.plugin.openshift.client.kubernetes.KubernetesResourceUtil.getRoutesByLabel(KubernetesResourceUtil.java:78)
  | at org.eclipse.che.plugin.openshift.client.OpenShiftDeploymentCleaner.cleanUpWorkspaceResources(OpenShiftDeploymentCleaner.java:47)
  | at org.eclipse.che.plugin.openshift.client.OpenShiftDeploymentCleaner.cleanDeploymentResources(OpenShiftDeploymentCleaner.java:36)
  | at org.eclipse.che.plugin.openshift.client.OpenShiftConnector.createContainer(OpenShiftConnector.java:540)
  | at org.eclipse.che.plugin.docker.machine.MachineProviderImpl.createContainer(MachineProviderImpl.java:643)
  | at org.eclipse.che.plugin.docker.machine.MachineProviderImpl.startService(MachineProviderImpl.java:352)
  | ... 13 common frames omitted
```

OSO account relinking via https://sso.openshift.io/auth/realms/fabric8/account/identity
does not seem to help

**Comment by ibuziuk** [comment]
@alexeykazakov any ideas ?

**Comment by ibuziuk** [comment]
@ldimaggi not sure if it is reproducible for other accounts, would be great if you could check

**Comment by alexeykazakov** [comment]
Looks like a problem with OS service account for che tenant? I don't see here anything related to OSIO or OSO user tokens.

**Comment by joshuawilson** [comment]
This was an AWS limit issue and has been fixed.

**Comment by ibuziuk** [comment]
I still have this problem after resetting environment:

> 2017-11-03 06:02:55,272[aceSharedPool-0]  [INFO ] [o.e.c.p.o.c.OpenShiftConnector 539]  - Failure executing: POST at: https://kubernetes.default.svc/oapi/v1/namespaces/ibuziuk-che/routes. Message: Route "wsagent-di3szm0qp4l3s8s4" is invalid: spec.host: Forbidden: you do not have permission to set the host field of the route. Received status: Status(apiVersion=v1, code=422, details=StatusDetails(causes=[StatusCause(field=spec.host, message=Forbidden: you do not have permission to set the host field of the route, reason=FieldValueForbidden, additionalProperties={})], group=null, kind=Route, name=wsagent-di3szm0qp4l3s8s4, retryAfterSeconds=null, additionalProperties={}), kind=Status, message=Route "wsagent-di3szm0qp4l3s8s4" is invalid: spec.host: Forbidden: you do not have permission to set the host field of the route, metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=Invalid, status=Failure, additionalProperties={}).

**Comment by joshuawilson** [comment]
:( ok thanks

**Comment by ibuziuk** [comment]
@alexeykazakov yeah, it is not related to external token, but rather service account one :/

**Comment by l0rd** [comment]
It looks like Che service account doesn't have the right to create a route anymore. @aslakknutsen may be related to https://github.com/fabric8-services/fabric8-tenant/commit/df0db9a77c70b9cf486cbbbec594e2042a8fa75b?

**Comment by l0rd** [comment]
Looking better at the problem it looks like the policy for route creation on OSO has changed yesterday. Route creation fails if the host is explicitly specified (was ignored before):

```bash
echo "apiVersion: v1
kind: Route
metadata:
  name: che3
spec:
  host: 8a09.starter-us-east-2.openshiftapps.com
  tls:
    insecureEdgeTerminationPolicy: Redirect
    termination: edge
  to:
    kind: Service
    name: che-host" | oc apply -f -
```
fails with error message "**The Route "che3" is invalid: spec.host: Forbidden: you do not have permission to set the host field of the route**". 

If `host` is not specified the route is created successfully.

**Comment by pbergene** [comment]
I did some testing around this and confirmed that this also applies to the service token that owns the namespace. After inquiring with ops I got a reply saying that there was a change done yesterday to how us-west-1 and us-west-2 handle routes and that custom route host are no longer allowed. It looks like this has also been applied to us-east-2. The previous behaviour seems to be that this field was ignored, but now it will fail on route creation that includes this field.
